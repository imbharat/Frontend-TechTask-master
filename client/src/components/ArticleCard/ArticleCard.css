.article {
  /* border: 1px solid lavenderblush; */
  padding: 10px;
}

.article:hover {
  box-shadow: 8px 8px 8px #9f9f9f;
  border-radius: 0 0 10px 0;
}

.article > * {
  display: inline-block;
  text-align: center;
  padding: 4px 0;
  margin: 4px 0;
  width: 100%;
}

.article > input[type="button"] {
  padding: 0.3em;
  background-color: #e2e2e2;
  border: 1px solid lightgray;
  border-radius: 30px;
  cursor: pointer;
}

.article > input[type="button"]:hover {
  background-color: white;
  font-weight: 600;
}

.article > .articleName {
  font-weight: 600;
  font-family: Arial, Helvetica, sans-serif;
}
