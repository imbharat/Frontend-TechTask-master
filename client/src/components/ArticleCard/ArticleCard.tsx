import React from 'react';
import { useDispatch } from 'react-redux';
import { Article } from '../../shared/types';
import { formatter } from '../../shared/helpers';
import { updateCart } from '../../redux/CartStore/CartActions';
import { Product } from '../../redux/CartStore/Types';
import './ArticleCard.css';

function ArticleCard({ article, articleId }: { article: Article, articleId: string }) {

  const product: Product = {
    id: articleId,
    quantity: 1,
    details: {
      name: article.name,
      image: article.images[0].path,
      price: article.prices.regular.value
    }
  }

  const dispatch = useDispatch();
  
  return (
    <div className={'article'}>
      <img src={article.images[0].path} alt='Failed To Load Item'/>
      <div data-testid='article-name' className="articleName">{article.name}</div>
      <div>
        {formatter.format(article.prices.regular.value / 100)}
      </div>
      <input type="button" value="Add to cart" onClick={() => dispatch(updateCart(product))}/>
    </div>
  );
}

export default React.memo(ArticleCard);
