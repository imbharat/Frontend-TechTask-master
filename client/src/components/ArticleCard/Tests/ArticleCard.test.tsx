import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ArticleCard from '../ArticleCard';
import { article } from './ArticleCardTestData';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch
}));

describe("ArticleCard", () => {
    //snapshot test
    test('Article renders', () => {
        //arrange
        const data = article;
        //act
        const tree = renderer.create(<ArticleCard article={data} articleId={data.name + 0}/>).toJSON();
        //assert
        expect(tree).toMatchSnapshot();
    });

    test('Article name exists', () => {
        //arrange
        const data = article
        //act
        render(<ArticleCard article={data} articleId={data.name + 0}/>);
        const articleName = screen.queryByTestId('article-name')?.textContent;
        //assert
        expect(data.name).toEqual(articleName);
    });
})