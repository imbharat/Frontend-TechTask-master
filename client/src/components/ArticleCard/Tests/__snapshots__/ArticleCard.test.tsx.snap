// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard Article renders 1`] = `
<div
  className="article"
>
  <img
    alt="Failed To Load Item"
    src="https://cdn1.home24.net/images/media/catalog/product/200x200/png/b/e/bett-eos-buche-massiv-geoelt-257225.webp"
  />
  <div
    className="articleName"
    data-testid="article-name"
  >
    Massivholzbett EosWOOD (inkl Bettkästen)
  </div>
  <div>
    529,99 €
  </div>
  <input
    onClick={[Function]}
    type="button"
    value="Add to cart"
  />
</div>
`;
